### DISCUSSION TITLE
JSON profile data type

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#4569] [Ideas]
We've been using JSON embedded in Profile fields to track additional user context like pages-visited.  When editing a profile, JSON data is seen as a blank string.  It would be nice to see it in a multi-line text editing region.  Do we need a new data-type for this?

**Comment by sbwalker** [comment]
@markdav-is User Profile fields are stored as Settings for each User. The general idea with Settings in Oqtane is to utilize them as a standard dictionary - ie. name/value pairs. So don't try to aggregate multiple fields into a single JSON value - just use a name/value pair for each individual field. This will allow you to reference the fields using PageState.User.Settings["FieldName"] in your components.

**Reply by markdav-is** [reply]
Sounds like I'm overloading that concept to track more data.  We are trying to track what page "progress" are attached to the user as a percentage.  Instead of using my own table, I was stuffing some JSON in the profile settings for easy access via PageState.User.Settings["Progress"].  It works, but obviously I'm going against the intention of the design.  The only thing messing with our plan was the Profile Editing page would eat the JSON.  I'm going to make a new table to track this stuff to be more in-line with the spirit of the Profiles vs Modules.

**Reply by sbwalker** [reply]
Profile properties do have the ability to specify a Rows value so that the value is rendered in a textarea element. It should also be noted that User Settings do not need to always be Profile properties - you can save arbitrary User Settings using whatever key name you want in your code - they just will not be visible in the User Profile UI
